File: 백준/Bronze/3062. 수 뒤집기/수 뒤집기.py
def is_palindrome(n):
    # 주어진 수가 대칭수인지 판별하는 함수
    return str(n) == str(n)[::-1]

# 테스트 케이스 개수 입력받기
T = int(input())

for _ in range(T):
    # 수 입력받기
    num = int(input())

    # 수를 뒤집은 값과 원래 값의 합 구하기
    reversed_num = int(str(num)[::-1])
    total = num + reversed_num

    # 합이 대칭수인지 판별하여 결과 출력하기
    if is_palindrome(total):
        print("YES")
    else:
        print("NO")

File: 백준/Bronze/3062. 수 뒤집기/수 뒤집기.java
import java.util.Scanner;

public class Main {
    public static boolean isPalindrome(int n) {
        // 주어진 수가 대칭수인지 판별하는 메서드
        String numStr = Integer.toString(n);
        String reversedNumStr = new StringBuilder(numStr).reverse().toString();
        return numStr.equals(reversedNumStr);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // 테스트 케이스 개수 입력받기
        int T = scanner.nextInt();

        for (int i = 0; i < T; i++) {
            // 수 입력받기
            int num = scanner.nextInt();

            // 수를 뒤집은 값과 원래 값의 합 구하기
            int reversedNum = Integer.parseInt(new StringBuilder(Integer.toString(num)).reverse().toString());
            int total = num + reversedNum;

            // 합이 대칭수인지 판별하여 결과 출력하기
            if (isPalindrome(total)) {
                System.out.println("YES");
            } else {
                System.out.println("NO");
            }
        }
    }
}